pulpcore/cli/core/group.py

```diff
@@ -193,13 +207,13 @@ def role(ctx: click.Context, pulp_ctx: PulpCLIContext, group_ctx: PulpGroupConte
         decorators=[
             group_option,
             click.option("--role", required=True),
-            click.option("--object", "content_object", required=True),
             click.option(
                 "--object",
                 "content_object",
-                required=True,
+                default="",
```

Comment:
It is required on purpose. Assigning global permissions should not be done by accident.

Reply:
Added a callback to keep current behavior if domain is not specified.